src/irc/core/irc-cap.c

```diff
@@ -0,0 +1,82 @@
+#include "module.h"
+#include "signals.h"
+#include "misc.h"
+
+#include "irc-cap.h"
+#include "irc-servers.h"
+
+void cap_register_done (IRC_SERVER_REC *server)
+{
+	/* We managed to get all the caps we requested */ 
+	if (++server->cap_got == g_hash_table_size(server->cap_req)) {
+		server->cap_complete = 1;
+		irc_send_cmd_now(server, "CAP END");
+	}
+}
+
+#define foreach_cap(t,l) \
+	for (t = strtok(l, " "); t; t = strtok(NULL, " "))
```

Comment:
strtok is kinda ugly (and very thread unsafe, not that irssi uses threads but still ugly).

And not sure if this macro is even needed. `g_strsplit()`?